apps/browser-extension-wallet/src/views/browser-view/features/sing-message/SignMessage.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react';
+import styles from './SignMessage.module.scss';
+import { useTranslation } from 'react-i18next';
+import { WalletOwnAddressDropdown, AddressSchema } from '@lace/core';
+import { Drawer, DrawerNavigation, TextArea, Button, toast, PostHogAction } from '@lace/common';
+import { Text } from '@input-output-hk/lace-ui-toolkit';
+import { Cip30DataSignature } from '@cardano-sdk/dapp-connector';
+import { SignMessagePassword } from '@views/browser/features/sing-message/SignMessagePassword';
+import { Password } from '@input-output-hk/lace-ui-toolkit/dist/design-system/password-box/uncontrolled-password-box-input.component';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { useAnalyticsContext } from '@providers';
+
+export type SignMessageProps = {
+  addresses: AddressSchema[];
+  visible?: boolean;
+  onClose: () => void;
+  onSign: (address: string, message: string, password: string) => void;
```

Comment:
We should avoid passing the password around. Instead, use the `useSecrets` hook in parent component to access it.
```suggestion
  onSign: (address: string, message: string) => void;
```